### PR TITLE
add citrix_waf package

### PR DESCRIPTION
**Pull request by efd6**
## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] Dashboard will be in a follow-up PR
- [ ] Native Citrix format extended data is not handled as I have been unable to find a specification for it. If this becomes available it will also be in a follow-up PR. From what I can see with other ingesters, people tell users to configure their Citrix device to send logs in CEF.

## Related issues

- Updates #1308

### CODE REVIEW

File: packages/citrix_waf/_dev/build/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Citrix Web App Firewall
+
+The Citrix Web App Firewall prevents security breaches, data loss, and possible unauthorized modifications to websites that access sensitive business or customer information. It does so by filtering both requests and responses, examining them for evidence of malicious activity, and blocking requests that exhibit such activity. Your site is protected not only from common types of attacks, but also from new, as yet unknown attacks. In addition to protecting web servers and websites from unauthorized access, the Web App Firewall protects against vulnerabilities in legacy CGI code or scripts, web frameworks, web server software, and other underlying operating systems.
+
+## Compatibility
+
+FIXME
```
[the remark this answers is not in the document]
Comment (efd6) [reply packages/citrix_waf/_dev/build/docs/README.md:7]: Done

File: packages/citrix_waf/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      ignore_missing: true
+  - geoip:
+      field: source.ip
+      target_field: source.geo
+      ignore_missing: true
+
+  - uri_parts:
+      field: event.url.original
+      target_field: event.url
+      ignore_failure: true
```
Comment (legoguy1000) [inline packages/citrix_waf/data_stream/log/elasticsearch/ingest_pipeline/default.yml:89]: same about the `url` not `event.url`??

File: packages/citrix_waf/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        - _tmp
+        - _conf
+      ignore_failure: true
+
+on_failure:
+  - remove:
+      field:
+        - _tmp
+        - _conf
+      ignore_failure: true
```
Comment (P1llus) [inline packages/citrix_waf/data_stream/log/elasticsearch/ingest_pipeline/default.yml:124]: Could you change this to `ignore_missing`?

File: packages/citrix_waf/changelog.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# newer versions go on top
+- version: "0.0.1"
```
[the remark this answers is not in the document]
Comment (efd6) [reply packages/citrix_waf/changelog.yml:2]: I will bump to beta until I have the dashboard in place — next PR.

File: packages/citrix_waf/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+      if: ctx?._tmp?.timestamp8601 != null
+      field: _tmp.timestamp8601
+      timezone: '{{{event.timezone}}}'
+      formats:
+        - ISO8601
+  - set:
+      field: _tmp.timestamp
+      if: ctx._tmp?.timestamp != null && ctx.citrix?.event_year != null
+      value: "{{{citrix.event_year}}} {{{_tmp.timestamp}}}"
+  - date:
```
[the remark this answers is not in the document]
Comment (efd6) [reply packages/citrix_waf/data_stream/log/elasticsearch/ingest_pipeline/default.yml:50]: I have seen a wide variety in the documentation and it is deeply underspecified.

File: packages/citrix_waf/data_stream/log/manifest.yml [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+          #    sxSmbIUfc2SGJGCJD4I=
+          #    -----END CERTIFICATE-----
+      - name: tcp_options
+        type: yaml
+        title: Custom TCP Options
+        multi: false
+        required: false
+        show_user: false
+        default: |
+          #max_connections: 1
```
Comment (andrewkroh) [inline packages/citrix_waf/data_stream/log/manifest.yml:142]: One setting recommend for both UDP and TCP is to include a `max_message_size`. IIRC the default value is like 20 MiB which is way too big and wastes memory.

**Review by andrewkroh — APPROVED** [review]

File: packages/citrix_waf/data_stream/log/manifest.yml (hunk 6 above)
Comment (efd6) [reply packages/citrix_waf/data_stream/log/manifest.yml:152]: The WAF messages are fairly small, so I've suggested 10kB and also added a `udp_options` field.

Comment (andrewkroh) [reply packages/citrix_waf/data_stream/log/manifest.yml:152]: Sounds good. I suggest having that option uncommented so most users are using it by default when they enable this.

File: packages/citrix_waf/data_stream/log/manifest.yml [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+
+      - name: udp_options
+        type: yaml
+        title: Custom UDP Options
+        multi: false
+        required: false
+        show_user: false
+        default: |
+          max_message_size: 10KiB
+          read_buffer_size: 10KiB
```
Comment (andrewkroh) [inline packages/citrix_waf/data_stream/log/manifest.yml:59]: I didn't find any evidence that this `read_buffer_size` parameter exists. Do you know where this is used in the code?

Comment (efd6) [reply packages/citrix_waf/data_stream/log/manifest.yml:59]: This is used in the UDP package. Apparently it is useless there too.

**Pull request merged**